app/src/main/java/app/revanced/manager/ui/screen/settings/update/UpdateScreen.kt

```diff
@@ -0,0 +1,224 @@
+    AlertDialog(
+        onDismissRequest = { vm.showInternetCheckDialog = false },
+        dismissButton = {
+            TextButton(
+                onClick = {
+                    vm.showInternetCheckDialog = false
+                }
+            ) {
+                Text(stringResource(id = R.string.cancel))
+            }
+        },
+        confirmButton = {
+            TextButton(
+                onClick = {
+                    vm.showInternetCheckDialog = false
+                    vm.downloadUpdate(true)
+                }
+            ) {
+                Text(stringResource(id = R.string.download))
+            }
+        },
+        title = {
+            Text(text = stringResource(id = R.string.download_update_confirmation))
+        },
+        icon = {
+            Icon(Icons.Outlined.Update, stringResource(id = R.string.update))
+        },
+        text = {
+            Text(stringResource(id = R.string.download_confirmation_metered))
+        }
+    )
+}
+
+@Composable
+private fun Header(vm: UpdateViewModel) {
+    Column(verticalArrangement = Arrangement.spacedBy(16.dp)) {
+        Text(
+            text = stringResource(vm.state.title),
+            style = MaterialTheme.typography.headlineMedium
+        )
+        if (vm.state == State.CAN_DOWNLOAD) {
+            Column {
+                Text(
+                    text = stringResource(
+                        id = R.string.current_version,
+                        BuildConfig.VERSION_NAME
+                    ),
+                    style = MaterialTheme.typography.bodyMedium,
+                    color = MaterialTheme.colorScheme.onSurfaceVariant
+                )
+                vm.changelog?.let { changelog ->
+                    Text(
+                        text = stringResource(
+                            id = R.string.new_version,
+                            changelog.version.replace("v", "")
+                        ),
+                        style = MaterialTheme.typography.bodyMedium,
+                        color = MaterialTheme.colorScheme.onSurfaceVariant
+                    )
+                }
+            }
+        } else if (vm.state == State.DOWNLOADING) {
+            LinearProgressIndicator(
+                progress = vm.downloadProgress,
+                modifier = Modifier.fillMaxWidth()
+            )
+            Text(
+                text =
+                "${vm.downloadedSize.div(1000000)} MB /  ${vm.totalSize.div(1000000)} MB (${
+                    vm.downloadProgress.times(
+                        100
+                    ).toInt()
+                }%)",
+                style = MaterialTheme.typography.bodyMedium,
+                color = MaterialTheme.colorScheme.outline,
+                modifier = Modifier.align(Alignment.CenterHorizontally)
+            )
+        }
+    }
+}
+
+@Composable
+private fun ColumnScope.Changelog(changelog: Changelog) {
+    val scrollState = rememberScrollState()
+    Column(
```

Comment:
Maybe use `Box` instead since there is only one "child".

Reply:
It seems like a Column is necessary for this, using Box does this:

![image](https://github.com/ReVanced/revanced-manager/assets/79272171/14012f0b-4e47-4f77-8d21-32198bb3d6b1)

Reply:
Alright. Column is fine then